### PR TITLE
Rust agent config parsing

### PR DESCRIPTION
**Pull request by mattiecnvr**
### Description

This does two things:
1) It parses the agent-specific configs for the relayer, validator, and scraper.
2) It removes the rigid structures that had been in place for parsing that were called out earlier and instead parses the serde json values directly.

This PR does not cut over to the new versions just yet. That can happen once we start integrating it into the testing and deployment pipelines.

### Drive-by changes

- Derive more was added and used where it made sense
- The old config macro was removed

### Related issues

- Progress on #2215 

### Backward compatibility

Yes

### Testing

None

### CODE REVIEW

File: rust/agents/relayer/src/relayer.rs [hunk 1]
```diff
@@ -44,9 +42,11 @@ struct ContextKey {
 }
 
 /// A relayer agent
+#[derive(AsRef)]
 pub struct Relayer {
     origin_chains: HashSet<HyperlaneDomain>,
     destination_chains: HashSet<HyperlaneDomain>,
+    #[as_ref]
```
Comment (daniel-savu) [inline rust/agents/relayer/src/relayer.rs:49]: this is so much cleaner!

File: rust/hyperlane-base/src/settings/loader/mod.rs [hunk 2]
```diff
@@ -121,7 +129,6 @@ where
             }
 
             println!("Error during deserialization, showing the config for debugging: {formatted_config}");
-
             err.context("Config deserialization error, please check the config reference (https://docs.hyperlane.xyz/docs/operators/agent-configuration/configuration-reference)")
```
Comment (daniel-savu) [inline rust/hyperlane-base/src/settings/loader/mod.rs:132]: This could be an `.or_else(...)` now right?

**Review by daniel-savu — APPROVED** [review]
Great to see the raw config types removed, happy to merge if we go with the trait impl rather than the `parse!` macro

File: rust/hyperlane-base/src/settings/parser/parse_pipe.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+///
+/// Converts this:
+/// ```ignore
+/// parse! {
+///     p(err)
+///     |> get_key("chains")?
+///     |> get_obj_iter()?
+///     |> filter(|(k, _)| filter.contains(*k))
+///     |> collect()
+/// }
```
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply rust/hyperlane-base/src/settings/parser/parse_pipe.rs:11]: Agreed, after some discussion this morning and some iteration I have a solution that I think you will prefer.

File: rust/hyperlane-base/src/settings/loader/mod.rs [hunk 4]
```diff
@@ -104,10 +115,7 @@ where
     };
 
     match Config::try_deserialize::<T>(config_deserializer) {
```
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply rust/hyperlane-base/src/settings/loader/mod.rs:117]: Yep

**Pull request merged**